--- a/node_modules/playwright-core/lib/server/electron/electron.js
+++ b/node_modules/playwright-core/lib/server/electron/electron.js
@@ -129,7 +129,7 @@
   }
   async launch(progress, options) {
     let app = void 0;
-    let electronArguments = ["--inspect=0", "--remote-debugging-port=0", ...options.args || []];
+    let electronArguments = ["--inspect=0", ...options.args || []];
     if (import_os.default.platform() === "linux") {
       const runningAsRoot = process.geteuid && process.geteuid() === 0;
       if (runningAsRoot && electronArguments.indexOf("--no-sandbox") === -1)
@@ -162,6 +162,7 @@
       electronArguments = [];
     }
     delete env.NODE_OPTIONS;
+    delete env.ELECTRON_RUN_AS_NODE;
     const { launchedProcess, gracefullyClose, kill } = await (0, import_processLauncher.launchProcess)({

--- a/node_modules/playwright-core/lib/server/electron/loader.js
+++ b/node_modules/playwright-core/lib/server/electron/loader.js
@@ -1,9 +1,12 @@
 "use strict";
 const { app } = require("electron");
 const { chromiumSwitches } = require("../chromium/chromiumSwitches");
-process.argv.splice(1, process.argv.indexOf("--remote-debugging-port=0"));
-for (const arg of chromiumSwitches()) {
-  const match = arg.match(/--([^=]*)=?(.*)/);
+// [Electron, -r, loader.js[, --no-sandbox], --inspect=0, ...args]
+process.argv.splice(1, process.argv.indexOf("--inspect=0"));
+app.commandLine.appendSwitch("remote-debugging-port", "0");
+for (const arg of chromiumSwitches()) {
+  const match = arg.match(/--([^=]*)=?(.*)/);
+  if (match) app.commandLine.appendSwitch(match[1], match[2]);
 }
